var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -3,130 +3,185 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+
+# TODO: respect the spack-profided compiler flags if any
+
+# TODO: add MKL and BLIS support
+# Attention! MKL and BLIS are NOT JUST drop-in replacements
+# for blas/lapack!
+
+# TODO: Check that spack-provided  compiler flags
+# are respected
+
+# Attention!
+# Dynamic linking *appears* to work fine, but
+# Elk is expected to be linked statically.
+
+# TODO: check if it is possible/reasonable to disable OpenMP
+# It doesn't appear to be this way, but check anyway
+
+
 from spack import *
 
 
 class Elk(MakefilePackage):
-    """An all-electron full-potential linearised augmented-plane wave
-    (FP-LAPW) code with many advanced features."""
-
-    homepage = 'http://elk.sourceforge.net/'
-    url      = 'https://sourceforge.net/projects/elk/files/elk-3.3.17.tgz'
-
-    version('3.3.17', sha256='c9b87ae4ef367ed43afc2d43eb961745668e40670995e8e24c13db41b7e85d73')
-
-    # Elk provides these libraries, but allows you to specify your own
-    variant('blas',   default=True,
-            description='Build with custom BLAS library')
-    variant('lapack', default=True,
-            description='Build with custom LAPACK library')
-    variant('fft',    default=True,
-            description='Build with custom FFT library')
-
-    # Elk does not provide these libraries, but allows you to use them
-    variant('mpi',    default=True,
-            description='Enable MPI parallelism')
-    variant('openmp', default=True,
-            description='Enable OpenMP support')
-    variant('libxc',  default=True,
-            description='Link to Libxc functional library')
-
-    depends_on('blas',   when='+blas')
-    depends_on('lapack', when='+lapack')
-    depends_on('fftw',   when='+fft')
-    depends_on('mpi@2:', when='+mpi')
-    depends_on('libxc',  when='+libxc')
-
-    # Cannot be built in parallel
+    """An all-electron full-potential linearised augmented-plane wave (LAPW)
+    code with many advanced features. Written originally at
+    Karl-Franzens-Universität Graz as a milestone of the EXCITING EU Research
+    and Training Network, the code is designed to be as simple as possible so
+    that new developments in the field of density functional theory (DFT) can
+    be added quickly and reliably. The code is freely available under the GNU
+    General Public License.
+    """
+
+    homepage = "https://elk.sourceforge.io/"
+    url      = "https://downloads.sourceforge.net/project/elk/elk-7.1.14.tgz"
+
+# FIXME: Add a list of GitHub accounts to
+# notify when the package is updated.
+# maintainers = ['github_user1', 'github_user2']
```

Comment:
The unit tests are failing because there is still boilerplate code like this in the package. Remove this to get the tests to pass.

Reply:
That much I see.  But CentOS failed with syntax error, not boilerplate FIXME. And I can't see what could be wrong at the reported location.